src/parquet_writer.h

```diff
@@ -0,0 +1,190 @@
+      return;
+    }
+
+    filepath = info[1].ToString().Utf8Value();
+
+    // Build schema
+    auto schema_obj = info[0].As<Napi::Object>();
+    auto keys = schema_obj.GetPropertyNames();
+    parquet::schema::NodeVector fields;
+    for (uint32_t i = 0; i < keys.Length(); i++) {
+      auto name = keys.Get(i).ToString().Utf8Value();
+      auto type_name = schema_obj.Get(name).ToObject().Get("type").ToString().Utf8Value();
+      auto converted_type = convertedTypeFromString(type_name);
+      auto logical_type = logicalTypeFromString(type_name);
+      columns.push_back(Column{name, logical_type});
+      fields.push_back(parquet::schema::PrimitiveNode::Make(
+        name,
+        parquet::Repetition::OPTIONAL,
+        logical_type,
+        converted_type));
+    }
+    schema = std::static_pointer_cast<parquet::schema::GroupNode>(
+      parquet::schema::GroupNode::Make("schema", parquet::Repetition::OPTIONAL, fields)
+    );
+  }
+
+  Napi::Value appendRow(const Napi::CallbackInfo& info) {
+    auto env = info.Env();
+    if (info.Length() < 1 || !info[0].IsObject()) {
+      Napi::TypeError::New(env, "row::Object expected").ThrowAsJavaScriptException();
+      return Napi::Boolean::New(env, false);
+    }
+
+    auto row = info[0].As<Napi::Object>();
+    for (auto& i : columns) {
+      switch (i.type) {
+      case parquet::Type::INT32:
+        os << row.Get(i.key).ToNumber().Int32Value();
+        break;
+
+      case parquet::Type::INT64:
+        os << row.Get(i.key).ToNumber().Int64Value();
+        break;
+
+      case parquet::Type::FLOAT:
+        os << row.Get(i.key).ToNumber().FloatValue();
+        break;
+
+      case parquet::Type::DOUBLE:
+        os << row.Get(i.key).ToNumber().DoubleValue();
+        break;
+
+      case parquet::Type::BOOLEAN:
+        os << row.Get(i.key).ToBoolean().Value();
+        break;
+
+      case parquet::Type::BYTE_ARRAY:
+        os << row.Get(i.key).ToString().Utf8Value();
+        break;
+
+      default:
+        os << "null";
+      }
+    }
+    os << parquet::EndRow;
+
+    return Napi::Boolean::New(env, true);
+  }
+
+  Napi::Value open(const Napi::CallbackInfo& info) {
+    // Open file
+    std::shared_ptr<arrow::io::FileOutputStream> out_file;
+    PARQUET_ASSIGN_OR_THROW(
+      out_file,
+      arrow::io::FileOutputStream::Open(filepath)
+    );
+
+    // Setup output stream
+    os = parquet::StreamWriter {parquet::ParquetFileWriter::Open(
+      out_file,
+      schema,
+      builder.build()
+    )};
+
+    return Napi::Boolean::New(info.Env(), true);
+  }
+
+  Napi::Value setRowGroupSize(const Napi::CallbackInfo& info) {
+    builder.max_row_group_length(info[0].ToNumber().Int64Value());
+    return Napi::Boolean::New(info.Env(), true);
+  }
+
+protected:
+
+  std::string filepath;
+  std::shared_ptr<parquet::schema::GroupNode> schema;
+  parquet::WriterProperties::Builder builder;
+  parquet::StreamWriter os;
+  std::vector<Column> columns;
```

Comment:
Class fields should not be in-between methods, they should be at the start of the class.